### PR TITLE
Control Access section improvements

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Moves basic auth and overview of external providers to Control Access index page, along with a bit more explanation of authentication vs authorization
- Divides LDAP, AD, and OIDC content into three separate documents
- Removes the current auth.md doc
- Adds note about API-specific auth in API overview
- Adds OIDC as an external provider option in several docs where it was missing (like our commercial features page)
- Updates all links throughout docs
- Corrections re: authenticate vs. authorize in RBAC doc
- Adds redirect for auth.md to Control Access index.md

## Motivation and Context
Closes ~https://github.com/sensu/sensu-docs/issues/2307~ https://github.com/sensu/sensu-docs/issues/2811

### CODE REVIEW

**Comment by hillaryfraley** [comment]
@cwjohnston this is ready for review -- I left it as a draft PR for now b/c I will still need to make these updates in 6.0 when approved.

File: content/sensu-go/6.1/operations/control-access/rbac.md [hunk 1]
```diff
@@ -4,24 +4,23 @@ linkTitle: "RBAC Reference"
...
 product: "Sensu Go"
 menu:
   sensu-go-6.1:
     parent: control-access
 ---
 
-Sensu role-based access control (RBAC) helps different teams and projects share a Sensu instance.
-RBAC allows you to manage user access and resources based on namespaces, groups, roles, and bindings.
+Sensu's role-based access control (RBAC) helps different teams and projects share a Sensu instance.
+RBAC allows you to authoriize and manage user access and resources based on namespaces, groups, roles, and bindings.
```
Comment (cwjohnston) [inline content/sensu-go/6.1/operations/control-access/rbac.md]:
```suggestion
RBAC allows you to specify actions users are allowed to take against resources, within  namespaces or across all namespaces, based on roles bound to the user or to one or more groups which they are a member of.
```

**Review by cwjohnston — APPROVED** [review]

File: content/sensu-go/6.1/operations/control-access/rbac.md [hunk 2]
```diff
@@ -4,24 +4,23 @@ linkTitle: "RBAC Reference"
 reference_title: "Role-based access control (RBAC)"
 type: "reference"
 description: "Sensu's role-based access control (RBAC) helps different teams and projects share a Sensu instance. RBAC allows you to manage user access and resources based on namespaces, groups, roles, and bindings. Read the reference doc to learn about RBAC."
```
Comment (cwjohnston) [inline content/sensu-go/6.1/operations/control-access/rbac.md]: Can we work `authorization` or `authorize` in here?

File: content/sensu-go/6.1/operations/control-access/_index.md [hunk 3]
```diff
@@ -5,20 +5,143 @@ product: "Sensu Go"
...
+## Authorization
+
+After you set up authentication, configure authorization via [role-based access control (RBAC)][4] to give those users permissions within Sensu.
+Sensu RBAC allows you to manage and access users and resources based on namespaces, groups, roles, and bindings.
+See [Create a read-only user][5] for an example.
+
+- **Namespaces** partition resources within Sensu.
+Sensu entities, checks, handlers, and other [namespaced resources][17] belong to a single namespace.
+- **Roles** create sets of permissions (like GET and DELETE) tied to resource types.
+**Cluster roles** apply permissions across namespaces and include access to [cluster-wide resources][18] like users and namespaces.
```
Comment (cwjohnston) [inline content/sensu-go/6.1/operations/control-access/_index.md]:
```suggestion
**Cluster roles** apply permissions across all namespaces and may include access to [cluster-wide resources][18] like users and namespaces. 
```

File: content/sensu-go/6.1/operations/control-access/_index.md [hunk 4]
```diff
@@ -5,20 +5,143 @@ product: "Sensu Go"
...
+After you set up authentication, configure authorization via [role-based access control (RBAC)][4] to give those users permissions within Sensu.
+Sensu RBAC allows you to manage and access users and resources based on namespaces, groups, roles, and bindings.
+See [Create a read-only user][5] for an example.
+
+- **Namespaces** partition resources within Sensu.
+Sensu entities, checks, handlers, and other [namespaced resources][17] belong to a single namespace.
+- **Roles** create sets of permissions (like GET and DELETE) tied to resource types.
+**Cluster roles** apply permissions across namespaces and include access to [cluster-wide resources][18] like users and namespaces. 
+- **Role bindings** assign a role to a set of users and groups within a namespace.
+**Cluster role bindings** assign a cluster role to a set of users and groups cluster-wide.
```
Comment (cwjohnston) [inline content/sensu-go/6.1/operations/control-access/_index.md]:
```suggestion
**Cluster role bindings** assign a cluster role to a set of users and groups across all namespaces.
```

File: content/sensu-go/6.1/operations/control-access/_index.md [hunk 5]
```diff
@@ -5,20 +5,143 @@ product: "Sensu Go"
...
+
+To enable permissions for external users and groups within Sensu, you can create a set of [roles, cluster roles][11], [role bindings, and cluster role bindings][13] that map to the usernames and group names in your authentication provider.
+
+After you configure the correct roles and bindings, users can log in to [sensuctl][36] and the [web UI][1] using a single-sign-on username and password (no prefix required).
```
Comment (cwjohnston) [inline content/sensu-go/6.1/operations/control-access/_index.md]:
```suggestion
After you configure an authentication provider and establish the roles and bindings to grant authenticated users the desired privileges, those users can log in via [sensuctl][36] and the [web UI][1] using a single-sign-on username and password (no prefix required).
```

File: content/sensu-go/6.1/operations/control-access/_index.md [hunk 6]
```diff
@@ -5,20 +5,143 @@ product: "Sensu Go"
...
+{{< code shell >}}
+sensuctl auth info openldap
+{{< /code >}}
+
+To delete an authentication provider named `openldap`:
+
+{{< code shell >}}
+sensuctl auth delete openldap
+{{< /code >}}
+
```
Comment (cwjohnston) [inline content/sensu-go/6.1/operations/control-access/_index.md:93]: Consider adding a high-level example of logging in via sensuctl to complete the circle. This gives is a place to add a note similar to `Users _do not_ need to provide the username_prefix when logging in to Sensu.` which is currently only documented for individual authentication providers.

File: content/sensu-go/6.1/operations/control-access/_index.md [hunk 7]
```diff
@@ -5,20 +5,143 @@ product: "Sensu Go"
...
+## Authentication
+
+Administrators can use Sensu's [built-in basic authentication provider][14] or configure [external authentication providers][15] to authenticate via Lightweight Directory Access Protocol (LDAP), Active Directory (AD), or OpenID Connect 1.0 protocol (OIDC).
+
+### Use built-in basic authentication
+
+Sensu's built-in basic authentication provider allows you to create and manage user credentials (usernames and passwords) with the [users API][53], either directly or using [sensuctl][2].
+The basic authentication provider does not depend on external services and is not configurable.
+
+Sensu records basic authentication credentials in [etcd][54].
```
Comment (cwjohnston) [inline content/sensu-go/6.1/operations/control-access/_index.md]:
```suggestion
Sensu hashes user passwords using the [bcrypt][26] algorithm and records the basic authentication credentials in [etcd][54].
```

link bcrypt to https://en.wikipedia.org/wiki/Bcrypt

File: content/sensu-go/6.1/operations/control-access/_index.md [hunk 8]
```diff
@@ -5,20 +5,143 @@ product: "Sensu Go"
...
+[15]: #use-an-authentication-provider
+[17]: rbac#namespaced-resource-types
+[18]: rbac#cluster-wide-resource-types
+[19]: ../maintain-sensu/troubleshoot#log-levels
+[21]: ldap-auth/#ldap-group-search-attributes
+[22]: ldap-auth/#ldap-user-search-attributes
+[23]: ad-auth/#ad-metadata-attributes
+[24]: ldap-auth/#ldap-metadata-attributes
+[25]: /oidc-auth/#oidc-spec-attributes
+[27]: ../../api/authproviders/
```
Comment (hillaryfraley) [inline content/sensu-go/6.1/operations/control-access/_index.md:137]:
```suggestion
[26]: https://en.wikipedia.org/wiki/Bcrypt
[27]: ../../api/authproviders/
```

**Pull request merged**